File: frontend/src/types/Movie.ts
import {Genre} from "@/types/Genre";
import {WatchingStatus} from "@/types/WatchingStatus";
import {Country} from "@/types/Country";
import {RatingDistribution} from "@/types/RatingDistribution";
import {WatchingStatusDistribution} from "@/types/WatchingStatusDistribution";

export type Movie = {
    watchingStatus: WatchingStatus | null,
    addedAt: string | null,
    description: string | null,
    duration: number | null,
    movieId: number | null,
    moviePath: string | null,
    cover: string | null,
    banner: string | null,
    rating: number | null,
    genres: Genre[] | null,
    releaseYear: number | null,
    title: string | null,
    originCountry: Country | null,
    mpaaRating: string | null,
    studio: string | null,
    ratingDistribution: RatingDistribution | null,
    watchingStatusDistribution: WatchingStatusDistribution | null,
    totalRates: number | null,
    totalWatchingStatuses: number | null,
}

File: backend/src/main/java/program/moviesappbackend/api/movies/models/Movie.java
package program.moviesappbackend.api.movies.models;

import lombok.Builder;
import lombok.Getter;
import lombok.Setter;
import program.moviesappbackend.api.filter.genres.Genre;

import java.sql.Timestamp;
import java.util.List;

@Getter
@Builder
@Setter
public class Movie {
    private long movieId;
    private String watchingStatus;
    private String title;
    private int duration;
    private double rating;
    private String description;
    private String cover;
    private String banner;
    private int releaseYear;
    private String moviePath;
    private Timestamp addedAt;
    private List<Genre> genres;
    private String originCountry;
    private String mpaaRating;
    private String studio;
    private List<RatingDistribution> ratingDistribution;
    private List<WatchingStatusDistribution> watchingStatusDistribution;
    private List<Person> actors;
    private List<Person> directors;
    private int userRating;
}
